lib/developer_portal/app/controllers/developer_portal/admin/applications_controller.rb

```diff
@@ -113,10 +113,12 @@ def applications
   end
 
   def service
-    @service ||= if not site_account.multiservice?
-                   site_account.services.default
-                 elsif service_id = params[:service_id]
-                   site_account.services.find_by_id_or_system_name(service_id)
+    @service ||= if !multiservice?
+                   services.default
+                 elsif current_account.services_can_create_app_on.count == 1
+                   services.first
+                 elsif (service_id = params[:service_id])
```

Comment:
I wonder if this condition shouldn't be the first one to check. 🤔

Reply:
I think there was a reason for this but I can't remember now so I'll move it first 😄

Reply:
Maybe the reason why `service_id` went last was so the user could not force it when multiple services is not enabled. I think it's safer not to touch this. Sorry for suggesting before.

Reply:
That must be it